pool.go

```diff
@@ -67,3 +82,28 @@ func (p *pool) Close() {
 	p.cond.L.Unlock()
 	p.cond.Broadcast()
 }
+
+func (p *pool) removeIdleConns() {
+	if p.idleConnTTL == nil {
+		return
+	}
+
+	for p.size > p.minSize {
+		w := p.list[0]
+		timer, ok := p.timers[w]
+		if !ok {
+			return
+		}
+
+		select {
+		case <-timer.C:
+			w.Close()
+			p.size--
+			p.list = p.list[1:]
```

Comment:
Doing this will make the underlying storage of the slice unable to be reclaimed. We should still remove connections from the tail.